pkg/gossiper/db.go

```diff
@@ -67,3 +68,33 @@ func (d *mydb) ListData() (int, []pair) {
 	}
 	return SUCCEED, ans
 }
+
+type mybatch struct {
+	batch *leveldb.Batch
+}
+
+func NewBatch() *mybatch {
+	return &mybatch{
+		batch: new(leveldb.Batch),
+	}
+}
+
+func (mb *mybatch) Set(keys, vals []string) {
+	for i := 0; i < len(keys); i++ {
+		mb.batch.Put([]byte(keys[i]), []byte(vals[i]))
+	}
+}
+
+func (mb *mybatch) Remove(keys []string) {
+	for i := 0; i < len(keys); i++ {
+		mb.batch.Delete([]byte(keys[i]))
+	}
+}
+
+func (d *mydb) WriteDown(mybatch *leveldb.Batch) int {
```

Comment:
Come on... rename this.

Reply:
You want to name the functions for easy interpretation so that people can guess the function name instead of looking at your code to know what is what.

Reply:
Maybe name it Commit